drivers/metal/metal_objects.mm

```diff
@@ -1082,7 +1083,7 @@
 
 	UniformSet const &set = p_shader->sets[index];
 
-	for (uint32_t i = 0; i < uniforms.size(); i++) {
+	for (uint32_t i = 0; i < std::min(uniforms.size(), set.uniforms.size()); i++) {
```

Comment:
```suggestion
	for (uint32_t i = 0; i < MIN(uniforms.size(), set.uniforms.size()); i++) {
```
This should use our math library instead of std

Reply:
That's weird, as `core/typedefs.h` which defines our `MIN` and `MAX` also `undef`'s the pre-existing ones.

Maybe you need an explicit `core/typedefs.h` include here? It's kind of taken for granted usually as most Godot headers include it one way or another, but maybe this specific file of the Metal driver doesn't as it's relatively self-contained?

Reply:
Unfortunately, I'd have to include `typedefs.h` before the system libraries, and then `NSObjCRuntime.h` re-defines it again. Quite a nuisance!

Reply:
This is what it looks like:

```c
#if !defined(MIN)
    #define __NSMIN_IMPL__(A,B,L) ({ __typeof__(A) __NSX_PASTE__(__a,L) = (A); __typeof__(B) __NSX_PASTE__(__b,L) = (B); (__NSX_PASTE__(__a,L) < __NSX_PASTE__(__b,L)) ? __NSX_PASTE__(__a,L) : __NSX_PASTE__(__b,L); })
    #define MIN(A,B) __NSMIN_IMPL__(A,B,__COUNTER__)
#endif
```